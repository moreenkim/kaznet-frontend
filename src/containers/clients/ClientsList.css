#header {
    text-align: center
}

#client_name {
    text-align: center
}
