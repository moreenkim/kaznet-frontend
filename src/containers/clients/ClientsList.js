import React, { Component } from "react";
import { connect } from "react-redux";
import * as clientActions from "../../store/clients/actions";
import * as clientSelectiors from "../../store/clients/reducer";
import * as globalActions from "../../store/global/actions";
import ListView from "../../components/ListView";
import ElementMap from "../ElementMap";

class ClientsList extends Component {
  componentDidMount() {
    this.props.dispatch(clientActions.fetchClients());
    this.props.dispatch(globalActions.changePageTitle("Clients"));
    this.props.dispatch(globalActions.changePageTitleButton("+ Create Client"));
  }

  render() {
    if (!this.props.rowsById) return this.renderLoading();
    return (
      <div className="ClientsList">
        <ListView
          renderHeaders={this.renderHeaders}
          rowsIdArray={this.props.rowsIdArray}
          rowsById={this.props.rowsById}
          renderRow={this.renderRow}
        />
      </div>
    );
  }

  renderLoading() {
    return <p>Loading...</p>;
  }

  renderHeaders() {
    const headerItems = ["Name"];
    return <ElementMap items={headerItems} HTMLTag="th" />;
  }

  renderRow(row) {
    const rowItems = [row.attributes.name];
    return <ElementMap items={rowItems} HTMLTag="td" />;
  }
}

function mapStateToProps(state) {
  return {
    rowsById: clientSelectiors.getClientsById(state),
    rowsIdArray: clientSelectiors.getClientsIdArray(state)
  };
}

export default connect(mapStateToProps)(ClientsList);
