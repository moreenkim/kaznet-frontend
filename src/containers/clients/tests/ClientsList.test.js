import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Router } from "react-router";
import createBrowserHistory from "history/createBrowserHistory";

import { ClientsList } from "../ClientsList";
import * as fixtures from "../../../store/clients/tests/fixtures";

const history = createBrowserHistory();

describe("containers/clients/ClientsList", () => {
  it("renders without crashing", () => {
    shallow(
      <ClientsList
        fetchClients={function() {}}
        changePageTitle={function() {}}
        changePageTitleButton={function() {}}
        changePageTarget={function() {}}
        showListTitle={function() {}}
        changePageNumber={function() {}}
        location={history.location}
        searchVal={function() {}}
        rowsIdArray={fixtures.clientsIdArray}
      />
    );
  });

  it("renders client list correctly", () => {
    const wrapper = mount(
      <Router history={history}>
        <ClientsList
          fetchClients={function() {}}
          changePageTitle={function() {}}
          changePageTitleButton={function() {}}
          changePageTarget={function() {}}
          showListTitle={function() {}}
          rowsById={fixtures.clientsById}
          rowsIdArray={fixtures.clientsIdArray}
          endpoint={"clients"}
          pageLinks={fixtures.pageLinks}
          totalPages={fixtures.totalPages}
          currentPage={fixtures.currentPage}
          firstPage={fixtures.firstPage}
          changePageNumber={function() {}}
          lastPage={fixtures.lastPage}
          searchVal={function() {}}
          searchParam={fixtures.searchParam}
          location={history.location}
        />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
