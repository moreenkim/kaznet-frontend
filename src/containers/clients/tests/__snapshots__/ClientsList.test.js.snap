// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/clients/ClientsList renders client list correctly 1`] = `
<ClientsList
  changePageTitle={[Function]}
  changePageTitleButton={[Function]}
  fetchClients={[Function]}
  rowsById={
    Object {
      "1": Object {
        "attributes": Object {
          "name": "Sol Knights",
        },
        "id": "1",
        "type": "Client",
      },
    }
  }
  rowsIdArray={
    Array [
      "1",
    ]
  }
>
  <div
    className="ClientsList"
  >
    <ListView
      renderHeaders={[Function]}
      renderRow={[Function]}
      rowsById={
        Object {
          "1": Object {
            "attributes": Object {
              "name": "Sol Knights",
            },
            "id": "1",
            "type": "Client",
          },
        }
      }
      rowsIdArray={
        Array [
          "1",
        ]
      }
    >
      <Table
        bordered={true}
        className="kaznet-table"
        responsiveTag="div"
        tag="table"
      >
        <table
          className="kaznet-table table table-bordered"
        >
          <thead>
            <tr>
              <ElementMap
                HTMLTag="th"
                items={
                  Array [
                    "Name",
                  ]
                }
              >
                <th
                  key="0"
                >
                  Name
                </th>
              </ElementMap>
            </tr>
          </thead>
          <tbody>
            <tr
              key="1"
            >
              <ElementMap
                HTMLTag="td"
                items={
                  Array [
                    "Sol Knights",
                  ]
                }
              >
                <td
                  key="0"
                >
                  Sol Knights
                </td>
              </ElementMap>
            </tr>
          </tbody>
        </table>
      </Table>
    </ListView>
  </div>
</ClientsList>
`;
